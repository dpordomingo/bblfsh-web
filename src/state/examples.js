import * as history from 'services/history';
import * as code from './code';
import * as languages from './languages';

import { java_example_1 } from 'examples/hello.java.js';
import { java_example_2 } from 'examples/swap.java.js';
import { python_example_1 } from 'examples/fizzbuzz.py.js';
import { python_example_2 } from 'examples/classdef.py.js';

export const DEFAULT_EXAMPLE = 'java_example_1';
const LANG_JAVA = 'java';
const LANG_PYTHON = 'python';

export const examples = {
  java_example_1: {
    name: 'hello.java',
    language: LANG_JAVA,
    code: java_example_1,
  },
  java_example_2: {
    name: 'swap.java',
    language: LANG_JAVA,
    code: java_example_2,
  },
  python_example_1: {
    name: 'fizzbuzz.py',
    language: LANG_PYTHON,
    code: python_example_1,
  },
  python_example_2: {
    name: 'classdef.py',
    language: LANG_PYTHON,
    code: python_example_2,
  },
};

export const initialState = {
  selected: '',
};

export const SET = 'bblfsh/examples/SET';
export const RESET = 'bblfsh/examples/RESET';

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case SET:
      return {
        ...state,
        selected: action.selected,
      };
    case RESET:
      return initialState;
    default:
      return state;
  }
};

export const select = key => dispatch => {
  if (!key) {
    return;
  }

  const example = examples[key];

  dispatch(code.set(example.name, example.code));
  dispatch(languages.set(example.language));

  // url side effect
  history.setExample();

  return dispatch({
    type: SET,
    selected: key,
  });
};

export const reset = () => ({ type: RESET });
