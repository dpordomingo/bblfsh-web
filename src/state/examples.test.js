import { reducer, initialState, SET, RESET } from './examples';

describe('examples/reducer', () => {
  it('SET', () => {
    expect(
      reducer(initialState, { type: SET, selected: 'java_example_1' })
    ).toMatchSnapshot();
  });

  it('RESET', () => {
    expect(
      reducer(initialState, {
        type: RESET,
      })
    ).toMatchSnapshot();
  });
});
