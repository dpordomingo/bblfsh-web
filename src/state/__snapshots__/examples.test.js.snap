// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`examples/reducer RESET 1`] = `
Object {
  "selected": "",
}
`;

exports[`examples/reducer SET 1`] = `
Object {
  "selected": "java_example_1",
}
`;
